builtin/credential/ldap/path_login.go

```diff
@@ -0,0 +1,87 @@
+package ldap
+
+import (
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathLogin(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: `login`,
+		Fields: map[string]*framework.FieldSchema{
+			"username": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "DN (distinguished name) to be used for login.",
+			},
+
+			"password": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Password for this user.",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.WriteOperation: b.pathLogin,
+		},
+
+		HelpSynopsis:    pathLoginSyn,
+		HelpDescription: pathLoginDesc,
+	}
+}
+
+func (b *backend) pathLogin(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+
+	policies, resp, err := b.Login(req, username, password)
+	if len(policies) == 0 {
+		return resp, err
+	}
+
+	sort.Strings(policies)
+
+	return &logical.Response{
+		Auth: &logical.Auth{
+			Policies: policies,
+			Metadata: map[string]string{
+				"username": username,
+				"password": password,
+				"policies": strings.Join(policies, ","),
+			},
+			DisplayName: username,
+		},
+	}, nil
+}
+
+func (b *backend) pathLoginRenew(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	username := req.Auth.Metadata["username"]
+	password := req.Auth.Metadata["password"]
```

Comment:
Uhm, I don't get it. This is an example of authentication:

```
↪ http POST http://127.0.0.1:8200/v1/auth/ldap/login/tesla password=password
HTTP/1.1 200 OK
Content-Length: 235
Content-Type: application/json
Date: Sat, 09 May 2015 20:41:53 GMT
Set-Cookie: token=d93d5bdc-e8cc-48bc-b1fa-cc7129a6f07f; Path=/; Expires=Sun, 08 May 2016 20:41:53 UTC

{
    "auth": {
        "client_token": "d93d5bdc-e8cc-48bc-b1fa-cc7129a6f07f",
        "lease_duration": 0,
        "metadata": {
            "policies": "root",
            "username": "tesla"
        },
        "policies": [
            "root"
        ],
        "renewable": false
    },
    "data": null,
    "lease_duration": 0,
    "lease_id": "",
    "renewable": false
}
```

I've tried to run `vault token-renew` like this:

```
↪ vault token-renew -address=http://127.0.0.1:8200 d93d5bdc-e8cc-48bc-b1fa-cc7129a6f07f
Error renewing token: Error making API request.

URL: PUT http://127.0.0.1:8200/v1/auth/token/renew/d93d5bdc-e8cc-48bc-b1fa-cc7129a6f07f
Code: 400. Errors:

* lease not found
```

The auth response JSON says that there is no lease and it's not renewable. I don't know how to make it return a correct lease; I've tried modifying the `LeaseOptions` during login like this:

```
	return &logical.Response{
		Auth: &logical.Auth{
			LeaseOptions: logical.LeaseOptions{
				Lease:     1 * time.Hour,
				Renewable: true,
			},
			Policies: policies,
                       [...]
```

but this doesn't work as well. Can you please help put me on the right track?